include/deal.II/base/utilities.h

```diff
@@ -443,22 +443,22 @@ namespace Utilities
    * be an integer type and the exponent @p iexp must not be negative.
    */
   template <typename T>
-  constexpr T
+  constexpr DEAL_II_HOST_DEVICE T
   pow(const T base, const int iexp)
   {
 #if defined(DEBUG) && !defined(DEAL_II_CXX14_CONSTEXPR_BUG)
     // Up to __builtin_expect this is the same code as in the 'Assert' macro.
     // The call to __builtin_expect turns out to be problematic.
-    if (!(iexp >= 0))
-      ::dealii::deal_II_exceptions::internals::issue_error_noreturn(
-        ::dealii::deal_II_exceptions::internals::ExceptionHandling::
-          abort_or_throw_on_exception,
-        __FILE__,
-        __LINE__,
-        __PRETTY_FUNCTION__,
-        "iexp>=0",
-        "ExcMessage(\"The exponent must not be negative!\")",
-        ExcMessage("The exponent must not be negative!"));
+    KOKKOS_IF_ON_HOST((if (!(iexp >= 0))::dealii::deal_II_exceptions::
+                         internals::issue_error_noreturn(
+                           ::dealii::deal_II_exceptions::internals::
+                             ExceptionHandling::abort_or_throw_on_exception,
+                           __FILE__,
+                           __LINE__,
+                           __PRETTY_FUNCTION__,
+                           "iexp>=0",
+                           "ExcMessage(\"The exponent must not be negative!\")",
+                           ExcMessage("The exponent must not be negative!"));))
```

Comment:
```suggestion
    KOKKOS_IF_ON_HOST(({if (!(iexp >= 0))::dealii::deal_II_exceptions::
                         internals::issue_error_noreturn(
                           ::dealii::deal_II_exceptions::internals::
                             ExceptionHandling::abort_or_throw_on_exception,
                           __FILE__,
                           __LINE__,
                           __PRETTY_FUNCTION__,
                           "iexp>=0",
                           "ExcMessage(\"The exponent must not be negative!\")",
                           ExcMessage("The exponent must not be negative!"));}))
```
should fix `clang-tidy`. That's at least how we do it in `include/deal.II/base/exceptions.h`.